Std/Tactic/TryThis.lean

```diff
@@ -76,6 +76,9 @@ partial def replaceMVarsByUnderscores [Monad m] [MonadQuotation m]
 def delabToRefinableSyntax (e : Expr) : TermElabM Term :=
   return ⟨← replaceMVarsByUnderscores (← delab e)⟩
 
+/-- The maximum width of an ideal line in source code, 100 is the current convention. -/
+def lineWidth : Nat := 100
```

Comment:
It's a bit annoying that we are replacing one [magic number](https://github.com/leanprover/lean4/blob/fb0d0245db9bcb9ff649794f57522c52cb80c993/src/Init/Data/Format/Basic.lean#L280) with another. Why not just use the [formatting option](https://github.com/leanprover/lean4/blob/fb0d0245db9bcb9ff649794f57522c52cb80c993/src/Lean/Data/Format.lean#L17-L20) that exists for the purpose?

Reply:
Well the default there is too wide for most suggestions, it's exactly `defWidth`. So I want to override that with 100 somehow and figured I may as well give it a name. Are you suggesting to use `w` equal to 100 if the option is unset and the option otherwise?
I don't want to change the default value completely as that seems way too invasive, and setting the option before calling format seems functionally the same as setting `w`

Reply:
I think we shouldn't be adding more magic numbers. If the default should be 100 for everything, that should be upstreamed to lean core. If the default is only 100 for mathlib, then they should set this as an option in the lakefile and the code here should honor that.

Reply:
I take your point about non-configurable magic numbers, I have pushed a commit that turns this into a user option at least.

I do however think that there are two numbers that we care about that have been conflated up till now.
The width of what we see in the tactic state, and separately the width of what we type into the editor.
Of course people can do what they want but the current status quo is that every time someone uses `simp?` in mathlib (a lot) they have to reflow the lines manually which is just annoying.
On the flipside I (and many others I'm sure) do find that having the tactic state be a bit wider than the "ruler" marking the maximum input width is nice, as most likely the ruler is not the rightmost part of people's editors (as we often go over 100 characters when writing so having the ruler at the edge of the screen is not ideal). So there would be a chunk of dead space in the tactic state if we limit that to exactly the desired input width.

As with web development there is not really a way to optimize these numbers without a survey of peoples set-ups and preferences, but I do think we shouldn't treat the two values as the same, and having them both (now) configurable options with sensible defaults is a good start

Reply:
> On the flipside I (and many others I'm sure) do find that having the tactic state be a bit wider than the "ruler" marking the maximum input width is nice, as most likely the ruler is not the rightmost part of people's editors (as we often go over 100 characters when writing so having the ruler at the edge of the screen is not ideal).

FWIW on my setup the 100 character line takes up about 55% of the screen, so with the sidebar also showing 100 characters I'm a bit over budget and have to frequently move things back and forth to see. So I don't have any "dead space" to speak of, and 100 + 120 is even more over budget.

Reply:
Well thats precisely my point about this being like web development! Without knowing peoples devices you cant really please everyone, hence the options!

FWIW i put the goal view under my code, it works great and i have loads of space ;)